library/src/main/java/com/google/android/material/motion/runtime/TargetScope.java

```diff
@@ -47,6 +47,7 @@
 
   private final SimpleArrayMap<Class<? extends Performer>, Performer> cache =
     new SimpleArrayMap<>();
+  private final SimpleArrayMap<String, Performer> namedCache = new SimpleArrayMap<>();
```

Comment:
this can be a `SAMap<String, NamedPlanPerformance>` type, which will help you below.